BlueprintUI/Sources/BlueprintView/BlueprintView.swift

```diff
@@ -220,6 +232,7 @@ public final class BlueprintView: UIView {
 
     public override func didMoveToWindow() {
         super.didMoveToWindow()
+        isVisible = (window != nil)
```

Comment:
I think we'll need a couple more checks here; also including the hidden-ness of the view and window, perhaps

Reply:
(Also, if any views upwards in the hierarchy are hidden this will be "wrong", I think? Eg trying to think of cases where this would fall through. I wonder a more robust approach would be to keep a record of if we were previously visible, and then if that changes during a layout subviews (eg compute the effective hidden-ness of the view again); perform the appropriate onAppear vs onDisappear in addition to when we're _removed_ from a window)

Reply:
> I think we'll need a couple more checks here; also including the hidden-ness of the view and window, perhaps

> (Also, if any views upwards in the hierarchy are hidden this will be "wrong", I think? Eg trying to think of cases where this would fall through. 

The idea of "visible" can get complex. What if a view is obscured? What if it has a very low alpha? I think it's OK for us to pick a definition, and not worry about chasing down every interpretation.

So, I could see checking this view's `isHidden`, but walking the ancestor chain seems like too big a lift, because we can't observe it. FWIW, SwiftUI's idea of visibility is not affected by the `hidden()` modifier.

> I wonder a more robust approach would be to keep a record of if we were previously visible, and then if that changes during a layout subviews (eg compute the effective hidden-ness of the view again); perform the appropriate onAppear vs onDisappear in addition to when we're _removed_ from a window)

I think this does what you're describing already? It tracks visibility and perform the callbacks whenever `isVisible` flips.

Reply:
Hmm ok, sounds good. I would maybe add the `hidden` check. I think maybe we'll see if adding additional knobs for sharp / unexpected edges is worth it based on how people use it!

Reply:
I personally don't like adding a `hidden` check, that property doesn't affect anything in the view hierarchy, and there's no way to be notified about it. I feel like it would be unexpected behavior if that affected these events. Maybe that's just me?

Reply:
👍